aten/src/ATen/Context.cpp

```diff
@@ -237,10 +241,24 @@ void Context::setBenchmarkLimitCuDNN(int b) {
 }
 
 bool Context::allowTF32CuBLAS() const {
+#ifdef USE_ROCM
+    const static auto allow_tf32 = c10::utils::check_env(hipblaslt_allow_tf32);
+    if (allow_tf32 != true) {
+      return false;
+    }
+#endif
   return float32_matmul_precision != at::Float32MatmulPrecision::HIGHEST;
 }
 
 void Context::setAllowTF32CuBLAS(bool b) {
+#ifdef USE_ROCM
+  const static auto allow_tf32 = c10::utils::check_env(hipblaslt_allow_tf32);
```

Comment:
I understand why you want to make this env var static.  But the unit tests where you set the env var and then try to unset it later will have different behavior depending on how the UTs are sharded and the order in which they're run.  The first test to call allowTF32CuBLAS() will be sticky and you'll never be able to change the env var setting later.

Reply:
Perhaps allowTF32CuBLAS is called infrequently enough to remove the static?  It's inside a USE_ROCM section so won't affect CUDA at all.

Reply:
sounds good. I guess in the future after TF32 support is released in rocm, we can check the GPU arch here.